ip/index.ts

```diff
@@ -812,7 +907,7 @@ export interface Options {
   /**
    * Trusted proxies.
    */
-  proxies?: ReadonlyArray<string | Cidr> | null | undefined;
+  proxies?: ReadonlyArray<Cidr | Service | string> | null | undefined;
```

Comment:
This API is still too complex. Nothing about the platform should leak into this API because a proxy is not the platform the application is *running* on.

In addition to `ReadonlyArray<Cidr | string>`,  `null`, and `undefined`, we should allow `Record<string, string>` that is a mapping of IPs or CIDRs to the header. To improve performance of the SDKs, we could also accept a pre-parsed Map as `Map<Cidr | string, string>`
```suggestion
  proxies?: ReadonlyArray<Cidr | string> | Record<string, string> | Map<Cidr | string, string> | null | undefined;
```

All other concepts for services can be pushed into other building blocks, such as my previous suggestion to map all Cloudflare ranges to the `cf-connecting-ip` header.